ouroboros-network-framework/src/Ouroboros/Network/ConnectionManager/Core.hs

```diff
@@ -1563,8 +1563,10 @@ withConnectionManager ConnectionManagerArguments {
                 return (DemoteToColdLocalNoop Nothing
                                               (abstractState $ Known connState))
 
-              InboundIdleState _connId _connThread _handle dataFlow ->
-                assert (dataFlow == Duplex) $
+              -- TODO: This assertion is benign and also hit rarely (once per
+              -- 100_000 simulations)
```

Comment:
Hi marcin!

Reply:
If the assertion fires, this means we are deregistering an outbound connection which is in `InboundIdleState`.  The `Server2` tests do not have a proper outbound governor, which can lead to this kind of problem.  It is benign, because if we try to unregistered a connection which already was unregistered, we do not need to do anything.  Still, it'd be nice to fix this, so we can have a stronger confidence that outbound governor / peer state actions, do not try to do these kind of things.

Reply:
https://github.com/input-output-hk/ouroboros-network/issues/3477